### PR TITLE
FrSky new sensors support

### PR DESCRIPTION
**Pull request by bsongis**
Don't merge it now, some other sensors are still missing

### CODE REVIEW

**Comment by 3djc** [comment]
I have added some more sensors. There is an issue with the gas flow sensor : we do not have flow unit. Adding a flow unit would require conversion (because if added after text, if would fail validity test < UNIT_FIRST_VIRTUAL in a couple of places)

File: radio/src/telemetry/frsky_sport.cpp [hunk 1]
```diff
@@ -220,6 +226,19 @@ void sportProcessTelemetryPacket(const uint8_t * packet)
           sportProcessTelemetryPacket(id, 0, instance, data & 0xffff);
           sportProcessTelemetryPacket(id, 1, instance, data >> 16);
         }
+        else if (id >= ESC_TEMPERATURE_FIRST_ID && id <= ESC_TEMPERATURE_LAST_ID) {
+          sportProcessTelemetryPacket(id, 0, instance, data & 0x00ff);
+        }
+        else if (id >= GASSUIT_TEMP_FIRST_ID && id <= GASSUIT_TEMP_LAST_ID) {
```
Comment (bsongis) [inline radio/src/telemetry/frsky_sport.cpp]: those 2 case can be combined (as I did for ESC), but perhaps the compiler does the optimization on his own

File: radio/src/telemetry/frsky_sport.cpp [hunk 2]
```diff
@@ -63,11 +63,17 @@ const FrSkySportSensor sportSensors[] = {
   { RBOX_CNSP_FIRST_ID, RBOX_CNSP_LAST_ID, 1, ZSTR_BATT2_CONSUMPTION, UNIT_MAH, 0 },
   { RBOX_STATE_FIRST_ID, RBOX_STATE_LAST_ID, 0, ZSTR_CHANS_STATE, UNIT_BITFIELD, 0 },
   { RBOX_STATE_FIRST_ID, RBOX_STATE_LAST_ID, 1, ZSTR_RB_STATE, UNIT_BITFIELD, 0 },
+  { SD1_FIRST_ID, SD1_LAST_ID, 0, ZSTR_SD1_CHANNEL, UNIT_RAW, 0 },
   { ESC_POWER_FIRST_ID, ESC_POWER_LAST_ID, 0, ZSTR_ESC_VOLTAGE, UNIT_VOLTS, 2 },
   { ESC_POWER_FIRST_ID, ESC_POWER_LAST_ID, 1, ZSTR_ESC_CURRENT, UNIT_AMPS, 2 },
   { ESC_RPM_CONS_FIRST_ID, ESC_RPM_CONS_LAST_ID, 0, ZSTR_ESC_RPM, UNIT_RPMS, 0 },
   { ESC_RPM_CONS_FIRST_ID, ESC_RPM_CONS_LAST_ID, 1, ZSTR_ESC_CONSUMPTION, UNIT_MAH, 0 },
   { ESC_TEMPERATURE_FIRST_ID, ESC_TEMPERATURE_LAST_ID, 0, ZSTR_ESC_TEMP, UNIT_CELSIUS, 0 },
+  { GASSUIT_TEMP_FIRST_ID, GASSUIT_TEMP_LAST_ID, 0, ZSTR_TEMP1, UNIT_CELSIUS, 0 },
```
Comment (bsongis) [inline radio/src/telemetry/frsky_sport.cpp]: I am not sure we can reuse ZSTR in multiple sensors here (unless we decide that GasSuite is not used with a temperature sensor)

File: radio/src/telemetry/frsky_sport.cpp [hunk 3]
```diff
@@ -63,6 +63,17 @@ const FrSkySportSensor sportSensors[] = {
   { RBOX_CNSP_FIRST_ID, RBOX_CNSP_LAST_ID, 1, ZSTR_BATT2_CONSUMPTION, UNIT_MAH, 0 },
   { RBOX_STATE_FIRST_ID, RBOX_STATE_LAST_ID, 0, ZSTR_CHANS_STATE, UNIT_BITFIELD, 0 },
   { RBOX_STATE_FIRST_ID, RBOX_STATE_LAST_ID, 1, ZSTR_RB_STATE, UNIT_BITFIELD, 0 },
+  { SD1_FIRST_ID, SD1_LAST_ID, 0, ZSTR_SD1_CHANNEL, UNIT_RAW, 0 },
+  { ESC_POWER_FIRST_ID, ESC_POWER_LAST_ID, 0, ZSTR_ESC_VOLTAGE, UNIT_VOLTS, 2 },
+  { ESC_POWER_FIRST_ID, ESC_POWER_LAST_ID, 1, ZSTR_ESC_CURRENT, UNIT_AMPS, 2 },
+  { ESC_RPM_CONS_FIRST_ID, ESC_RPM_CONS_LAST_ID, 0, ZSTR_ESC_RPM, UNIT_RPMS, 0 },
+  { ESC_RPM_CONS_FIRST_ID, ESC_RPM_CONS_LAST_ID, 1, ZSTR_ESC_CONSUMPTION, UNIT_MAH, 0 },
+  { ESC_TEMPERATURE_FIRST_ID, ESC_TEMPERATURE_LAST_ID, 0, ZSTR_ESC_TEMP, UNIT_CELSIUS, 0 },
+  { GASSUIT_TEMP_FIRST_ID, GASSUIT_TEMP_LAST_ID, 0, ZSTR_TEMP1, UNIT_CELSIUS, 0 },
```
Comment (bsongis) [inline radio/src/telemetry/frsky_sport.cpp]: I am not sure we can reuse ZSTR in multiple sensors here (unless we decide that GasSuite is not used with a temperature sensor)

File: radio/src/telemetry/frsky_sport.cpp [hunk 4]
```diff
@@ -210,6 +221,24 @@ void sportProcessTelemetryPacket(const uint8_t * packet)
...
         }
+        else if (id >= ESC_POWER_FIRST_ID && id <= ESC_RPM_CONS_LAST_ID) {
+          // 2 sensors (POWER and RPM_CONS)
+          sportProcessTelemetryPacket(id, 0, instance, data & 0xffff);
+          sportProcessTelemetryPacket(id, 1, instance, data >> 16);
+        }
+        else if (id >= ESC_TEMPERATURE_FIRST_ID && id <= ESC_TEMPERATURE_LAST_ID) {
+          sportProcessTelemetryPacket(id, 0, instance, data & 0x00ff);
+        }
+        else if (id >= GASSUIT_TEMP_FIRST_ID && id <= GASSUIT_TEMP_LAST_ID) {
```
Comment (bsongis) [inline radio/src/telemetry/frsky_sport.cpp]: those 2 cases can be combined (as I did for ESC), but perhaps the compiler does the optimization on his own
in fact, the comment is right, there are 2 sensors but you didn't combine them

**Comment by bsongis** [comment]
Thanks again for your help @3djc  this seems perfect to me, we just need some sensors to test the feature. I will try to have a sensors set for you

**Pull request merged**